crates/worker/src/codec.rs

```diff
@@ -0,0 +1,38 @@
+use js_sys::Uint8Array;
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::JsValue;
+
+/// Message Encoding and Decoding Format
+pub trait Codec {
+    /// Encode an input to JsValue
+    fn encode<I>(input: I) -> JsValue
+    where
+        I: Serialize;
+
+    /// Decode a message to a type
+    fn decode<O>(input: JsValue) -> O
+    where
+        O: for<'de> Deserialize<'de>;
+}
+
+/// Default message encoding with [bincode].
+#[derive(Debug)]
+pub struct Bincode {}
```

Comment:
Should bincode be the default, with no way to opt out? Postcard recently hit 1.0, which is no_std so it should have a smaller footprint in the binary. I think we should allow users to customize it. A simple feature flag (enabled by default) will be enough here. Those who want to customize can implement `Codec` manually

What do you think?